src/main/java/org/fcrepo/spec/testsuite/crud/StateToken.java

```diff
@@ -0,0 +1,198 @@
+        assertEquals("New " + STATE_TOKEN + "should not match previous token value", value,
+                     doHead(resourceUri).header(STATE_TOKEN));
+    }
+
+    /**
+     * 3.10.2-A
+     */
+    @Test(groups = {"MAY"})
+    public void badStateTokenOnPatchWhenStateTokensIgnored() {
+        final TestInfo info = setupTest("3.10.2-A",
+                                        "A client may include the X-If-State-Token header field in a PATCH request to" +
+                                        " make the request conditional on the resource's current state token matching" +
+                                        " the client's value. If an implementation does not support state tokens, it " +
+                                        "may ignore any X-If-State-Token header in HTTP PATCH requests.",
+                                        "https://fcrepo.github.io/fcrepo-specification/#x-state-token", ps);
+        final Response createResponse = createBasicContainer(uri, info.getId());
+        final String resourceUri = getLocation(createResponse);
+        final Response getResponse = doGet(resourceUri);
+        if (!isStateTokenSupported(getResponse)) {
+            final String sparqlUpdate =
+                "INSERT { <> <http://purl.org/dc/elements/1.1/title> \"test\" . } WHERE {}";
+            doPatchUnverified(resourceUri, new Headers(new Header(IF_STATE_TOKEN, "random-token"),
+                                                       new Header("Content-Type", "application/sparql-update")),
+                              sparqlUpdate).then().statusCode(successRange());
+        }
+    }
+
+    /**
+     * 3.10.2-B
+     */
+    @Test(groups = {"MUST"})
+    public void badStateTokenOnPatchWhenStateTokensSupported() {
+        final TestInfo info = setupTest("3.10.2-B",
+                                        "A client may include the X-If-State-Token header field in a PATCH request to" +
+                                        " make the request conditional on the resource's current state token matching" +
+                                        " the client's value.  An HTTP " +
+                                        "PATCH request that includes an X-If-State-Token header must be " +
+                                        "rejected with a 412 (Precondition Failed) response if the implementation " +
+                                        "supports state tokens, but the client-supplied value does not match the " +
+                                        "resource's current state token.",
+                                        "https://fcrepo.github.io/fcrepo-specification/#x-state-token", ps);
+        final Response createResponse = createBasicContainer(uri, info.getId());
+        final String resourceUri = getLocation(createResponse);
+        final Response getResponse = doGet(resourceUri);
+        checkStateTokensSupported(getResponse);
```

Comment:
Please add newlines and a comment around the point where the test may be skipped so that it is clear that the test may terminate at this point.